server/sonar-db-dao/src/main/java/org/sonar/db/MyBatis.java

```diff
@@ -0,0 +1,301 @@
+    confBuilder.loadAlias("RuleParam", RuleParamDto.class);
+    confBuilder.loadAlias("Rule", RuleDto.class);
+    confBuilder.loadAlias("SchemaMigration", SchemaMigrationDto.class);
+    confBuilder.loadAlias("ScrapProperty", ScrapPropertyDto.class);
+    confBuilder.loadAlias("Snapshot", SnapshotDto.class);
+    confBuilder.loadAlias("UserGroup", UserGroupDto.class);
+    confBuilder.loadAlias("UserPermission", UserPermissionDto.class);
+    confBuilder.loadAlias("UserTokenCount", UserTokenCount.class);
+    confBuilder.loadAlias("UserToken", UserTokenDto.class);
+    confBuilder.loadAlias("User", UserDto.class);
+    confBuilder.loadAlias("UuidWithProjectUuid", UuidWithProjectUuidDto.class);
+    confBuilder.loadAlias("ViewsSnapshot", ViewsSnapshotDto.class);
+
+    // ResourceMapper has to be loaded before IssueMapper because this last one used it
+    confBuilder.loadMapper(ResourceMapper.class);
+
+    // keep them sorted alphabetically
+    Class<?>[] mappers = {
+      ActiveRuleMapper.class,
+      AuthorMapper.class,
+      AuthorizationMapper.class,
+      CeActivityMapper.class,
+      CeQueueMapper.class,
+      CeScannerContextMapper.class,
+      CeTaskInputMapper.class,
+      ComponentKeyUpdaterMapper.class,
+      ComponentLinkMapper.class,
+      ComponentMapper.class,
+      CustomMeasureMapper.class,
+      DuplicationMapper.class,
+      EventMapper.class,
+      FileSourceMapper.class,
+      GroupMapper.class,
+      GroupMembershipMapper.class,
+      GroupPermissionMapper.class,
+      InternalPropertiesMapper.class,
+      IsAliveMapper.class,
+      IssueChangeMapper.class,
+      IssueMapper.class,
+      LoadedTemplateMapper.class,
+      MeasureMapper.class,
+      MetricMapper.class,
+      NotificationQueueMapper.class,
+      OrganizationMapper.class,
+      PermissionTemplateCharacteristicMapper.class,
+      PermissionTemplateMapper.class,
+      ProjectQgateAssociationMapper.class,
+      PropertiesMapper.class,
+      PurgeMapper.class,
+      QProfileChangeMapper.class,
+      QualityGateConditionMapper.class,
+      QualityGateMapper.class,
+      QualityProfileMapper.class,
+      RoleMapper.class,
+      RuleMapper.class,
+      RuleRepositoryMapper.class,
+      SchemaMigrationMapper.class,
+      SnapshotMapper.class,
+      UserGroupMapper.class,
+      UserMapper.class,
+      UserPermissionMapper.class,
+      UserTokenMapper.class,
+      WebhookDeliveryMapper.class
+    };
+    confBuilder.loadMappers(mappers);
+
+    sessionFactory = new SqlSessionFactoryBuilder().build(confBuilder.build());
+    return this;
+  }
+
+  public SqlSessionFactory getSessionFactory() {
+    return sessionFactory;
+  }
+
+  /**
+   * @deprecated since 4.4. Replaced by <code>openSession(false)</code>.
+   */
+  @Deprecated
+  public SqlSession openSession() {
+    return openSession(false);
+  }
+
+  /**
+   * @since 4.4
+   */
+  public DbSession openSession(boolean batch) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "batch" to determine which action to take. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)